code/game/alternate_appearance.dm

```diff
@@ -0,0 +1,157 @@
+/*
+	Alternate Appearances!
+	A framework for replacing an atom (and it's overlays) with an override = 1 image, that's less shit!
+	Example uses:
+		* hallucinating all mobs looking like skeletons
+		* people wearing cardborg suits appearing as Standard Cyborgs to the other Silicons
+		* !!use your imagination!!
+
+*/
+
+
+//This datum is built on-the-fly by some of the procs below
+//no need to instantiate it
+/datum/alternate_appearance
+	var/key = ""
+	var/image/img
+	var/list/viewers = list()
+	var/atom/owner = null
+
+
+/*
+	Displays the alternate_appearance
+	displayTo - a list of MOBS to show this appearance to
+*/
+/datum/alternate_appearance/proc/display_to(list/displayTo)
+	if(!displayTo || !displayTo.len)
+		return
+	for(var/m in displayTo)
+		var/mob/M = m
+		if(!M.viewing_alternate_appearances)
+			M.viewing_alternate_appearances = list()
+		viewers |= M
+		M.viewing_alternate_appearances[key] = src
+		if(M.client)
+			M.client.images |= img
+
+/*
+	Hides the alternate_appearance
+	hideFrom - optional list of MOBS to hide it from the list's mobs specifically
+*/
+/datum/alternate_appearance/proc/hide(list/hideFrom)
+	var/list/hiding = viewers
+	if(hideFrom)
+		hiding = hideFrom
+
+	for(var/m in hiding)
+		var/mob/M = m
+		if(M.client)
+			M.client.images -= img
+		if(M.viewing_alternate_appearances && M.viewing_alternate_appearances.len)
+			M.viewing_alternate_appearances[key] = null
+			M.viewing_alternate_appearances -= key
+			if(!M.viewing_alternate_appearances.len)
+				M.viewing_alternate_appearances = null
+		viewers -= M
+
+
+/*
+	Removes the alternate_appearance from its owner's alternate_appearances list, hiding it also
+*/
+/datum/alternate_appearance/proc/remove()
+	hide()
+	if(owner && owner.alternate_appearances)
+		owner.alternate_appearances[key] = null
```

Comment:
This line is unnecessary since the reference will be removed during the -= key below. Same applies in hide() above.